### PR TITLE
Refine the URL»pathname summary

### LINKED ISSUE #22333: URL.pathname ambiguous: does it permit or not permit the empty string?
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname

### What specific section or headline is this issue about?

URL.pathname

### What information was incorrect, unhelpful, or incomplete?

It is unclear whether you mean (A):

> The pathname property of the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) interface is 
(a string containing an initial / followed by the path of the URL, not including the query string or fragment) 
(or the empty string if there is no path).

or (B):

> The pathname property of the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) interface is 
a string containing an initial / followed by 
(the path of the URL, not including the query string or fragment (or the empty string if there is no path)).

### What did you expect to see?

Clarification as to whether the pathname must always begin with a leading `/`, or if it permits the empty string.

In reading (A) above, it permits the empty string. In reading (B) above, it does **not** permit the empty string.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/url/pathname`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/url/pathname/index.md
* Last commit: https://github.com/mdn/content/commit/f7dae62645a2c735ed6f6ed63f664bf279fdfc4b
* Document last modified: 2022-09-09T05:19:45.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/22333 ⬩ See https://github.com/mdn/content/pull/22335#discussion_r1024177100

### CODE REVIEW

File: files/en-us/web/api/url/pathname/index.md [hunk 1]
```diff
@@ -13,9 +13,11 @@ browser-compat: api.URL.pathname
...
+The **`pathname`** property of the {{domxref("URL")}} interface represents a location in a hierarchical structure. It is a string constructed from a list of path segments, each of which is prefixed by a `/` character. If the URL has no path segments, the value of its `pathname` property will be the empty string.
 
-The path of the URL is a `/` delimited sequence of segments, which may each be one of `.`, `..` or a string with zero or more characters excluding `/`, `?`, and `#`.
+URLs such as `https` and `http` URLs that have [hierarchical schemes](https://datatracker.ietf.org/doc/html/draft-fielding-url-syntax-09) (which the current URL standard calls “[special schemes](https://url.spec.whatwg.org/#special-scheme)”), always have at least one (invisible) path segment: the empty string — and thus the `pathname` value for such “[special scheme](https://url.spec.whatwg.org/#special-scheme)” URLs can never be the empty string but will instead always have a least one `/` character.
```
Comment (Josh-Cena) [inline files/en-us/web/api/url/pathname/index.md:18]:
```suggestion
URLs such as `https` and `http` URLs that have [hierarchical schemes](https://datatracker.ietf.org/doc/html/draft-fielding-url-syntax-09) (which the current URL standard calls "[special schemes](https://url.spec.whatwg.org/#special-scheme)") always have at least one (invisible) path segment: the empty string. Thus the `pathname` value for such "special scheme" URLs can never be the empty string, but will instead always have a least one `/` character.
```

File: files/en-us/web/api/url/pathname/index.md [hunk 2]
```diff
@@ -13,9 +13,11 @@ browser-compat: api.URL.pathname
...
+
+For example, the URL `https://developer.mozilla.org` has just one path segment: the empty string — so its `pathname` value is constructed by prefixing a `/` character to the empty string.
```
Comment (Josh-Cena) [inline files/en-us/web/api/url/pathname/index.md:20]:
```suggestion
For example, the URL `https://developer.mozilla.org` has just one path segment: the empty string, so its `pathname` value is constructed by prefixing a `/` character to the empty string.
```

**Review by Josh-Cena — COMMENTED** [review]
Now I understand better. Thanks!

File: files/en-us/web/api/url/pathname/index.md (hunk 2 above)
Comment (Josh-Cena) [reply files/en-us/web/api/url/pathname/index.md:20]: Also, I might give an additional case where you _can_ get the empty string, like `new URL("web:")`, but I've never used non-`file`, non-`https?` URLs, so I don't know the problem space very well.

Comment (sideshowbarker) [reply files/en-us/web/api/url/pathname/index.md:20]: > Also, I might give an additional case where you _can_ get the empty string, like `new URL("web:")`, but I've never used non-`file`, non-`https?` URLs, so I don't know the problem space very well.

Yeah, I had thought about that too — but the problem is, I could not come up with a real-world case of such a URL that would have the empty string as its pathname. Any case I can think of where you’d actually be identifying some real resource of some kind, it’s gonna end up having a pathname, as far as I can see.

**Review by Josh-Cena — APPROVED** [review]
Yeah, this makes sense to me as-is. I'm approving because I like it, but I'll also leave it there since I'm not a web API reviewer.

File: files/en-us/web/api/url/pathname/index.md [hunk 3]
```diff
@@ -13,9 +13,11 @@ browser-compat: api.URL.pathname
...
+The **`pathname`** property of the {{domxref("URL")}} interface represents a location in a hierarchical structure. It is a string constructed from a list of path segments, each of which is prefixed by a `/` character. If the URL has no path segments, the value of its `pathname` property will be the empty string.
 
-The path of the URL is a `/` delimited sequence of segments, which may each be one of `.`, `..` or a string with zero or more characters excluding `/`, `?`, and `#`.
+URLs such as `https` and `http` URLs that have [hierarchical schemes](https://datatracker.ietf.org/doc/html/draft-fielding-url-syntax-09) (which the current URL standard calls "[special schemes](https://url.spec.whatwg.org/#special-scheme)") always have at least one (invisible) path segment: the empty string. Thus the `pathname` value for such "special scheme" URLs can never be the empty string, but will instead always have a least one `/` character.
```
Comment (Elchi3) [inline files/en-us/web/api/url/pathname/index.md:18]:
```suggestion
URLs such as `https` and `http` URLs that have [hierarchical schemes](https://datatracker.ietf.org/doc/html/draft-fielding-url-syntax-09) (which the URL standard calls "[special schemes](https://url.spec.whatwg.org/#special-scheme)") always have at least one (invisible) path segment: the empty string. Thus the `pathname` value for such "special scheme" URLs can never be the empty string, but will instead always have a least one `/` character.
```

**Review by Elchi3 — APPROVED** [review]
Thanks! 👍 
Just one nit

**Pull request merged**